BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/cryptography/keys/PrivateKey.java

```diff
@@ -0,0 +1,34 @@
+package nl.tudelft.ewi.ds.bankchain.cryptography.keys;
+
+/**
+ * The PrivateKey class.
+ * Private keys allow to sign a message verifiable by the public key.
+ * Also allows to decrypt any message encrypted by the public key.
+ */
+public abstract class PrivateKey extends Key {
+
+    /**
+     * The public key that belongs to this private key.
+     */
+    private PublicKey publicKey;
+
```

Comment:
So that the implementer is swayed to not generate the public key everytime getPublicKey() is called, but store it after the first call.